setup.py

```diff
@@ -1,102 +1,188 @@
-    f.close()
-
-APPNAME = 'CellProfiler'
-APP = ['CellProfiler.py']
-icon_src_path = os.path.join('.', 'artwork')
-DATA_FILES = [('artwork',
-               glob.glob(os.path.join(icon_src_path, '*.png'))+
-               [os.path.join(icon_src_path, "icon_copyrights.txt")])]
-from javabridge import JARS
-imagej_path = os.path.abspath(os.path.join(".", "imagej", "jars"))
-jars = JARS + [os.path.join(imagej_path, jar) for jar in get_cellprofiler_jars()]
-jars.append(os.path.join(
-    imagej_path, "cellprofiler-java-dependencies-classpath.txt"))
-DATA_FILES.append(('imagej/jars', jars))
-OPTIONS = {'argv_emulation': True,
-           'packages': ['cellprofiler', 'contrib', 'imagej', 'javabridge'],
-           'includes': ['numpy', 'scipy', 'sklearn', 'sklearn.utils.sparsetools.*',
-                        'wx', 'matplotlib','email.iterators', 'smtplib', 'zmq',
-                        'javabridge', 'bioformats', 
-                        'sqlite3', 'libtiff', 'wx.lib.intctrl', 'libtiff.'+tiff_h_name,
-                        'xml.dom.minidom', 'h5py', 'h5py.defs', 'h5py.utils', 'h5py._proxy', 'readline'],
-           'excludes': ['pylab', 'Tkinter', 'Cython', 'scipy.weave',
-                        'virtualenv'],
-           'resources': ['artwork/CellProfilerIcon.png'],
-           'iconfile' : 'artwork/CellProfilerIcon.icns',
-           'frameworks' : [tiff_dylib, jpeg_dylib],
-           'plist': { 
-               "LSArchitecturePriority": ["i386"],
-               "LSMinimumSystemVersion": "10.6.8", # See #871
-               "CFBundleName": "CellProfiler",
-               "CFBundleIdentifier": "org.cellprofiler.CellProfiler",
-               "CFBundleShortVersionString": cellprofiler.utilities.version.dotted_version,
-               "CFBundleDocumentTypes": [{
-                   "CFBundleTypeExtensions":["cpproj"],
-                   "CFBundleTypeIconFile":"artwork/CellProfilerIcon.icns",
-                   "CFBundleTypeName":"CellProfiler project",
-                   "CFBundleTypeRole":"Editor"
-                   }, {
-                       "CFBundleTypeExtensions":["cppipe"],
-                       "CFBundleTypeIconFile":"artwork/CellProfilerIcon.icns",
-                       "CFBundleTypeName":"CellProfiler pipeline",
-                       "CFBundleTypeRole":"Editor"
-                       }]
-           }
-           }
-
-if sys.argv[-1] == 'py2app':
-    assert not os.path.exists("build"), "Remove the build and dist directories before building app!"
-    assert not os.path.exists("dist"), "Remove the build and dist directories before building app!"
-
-setup(
-    app=APP,
-    package_data={'javabridge':['jars/*.jar']},
-    data_files=DATA_FILES,
-    options={'py2app': OPTIONS},
-    setup_requires=['py2app'],
-    name="CellProfiler"
-)
+import setuptools.command.install
+
+setuptools.dist.Distribution({
+    "setup_requires": [
+        "clint",
+        "requests",
+        "pytest"
+    ]
+})
+
+
```

Comment:
My first guess about why `python setup.py build install` differs from `python setup.py bdist; pip install ..` is that neither the bdist command nor the pip install run the install class below. First step is to confirm that and if so, I will take a shot at making the download step happen as a subcommand of 'build' so that the jar is in place before bdist. Just a guess at this point and the change to the code may be pretty minimal.